src/almanack/metrics/data.py

```diff
@@ -115,6 +115,44 @@ def file_exists_in_repo(
     return False
 
 
+def is_citable(repo: pygit2.Repository) -> bool:
+    """
+    Check if the given repository is citable.
+
+    A repository is considered citable if it contains a CITATION.cff or CITATION.bib
+    file, or if the README.md file contains a citation section indicated by "## Citation"
+    or "## Citing".
+
+    Args:
+        repo (pygit2.Repository): The repository to check for citation files.
+
+    Returns:
+        bool: True if the repository is citable, False otherwise.
+    """
+
+    # Check for a CITATION.cff or CITATION.bib file
+    if file_exists_in_repo(
+        repo=repo,
+        expected_file_name="citation",
+        check_extension=True,
+        extensions=[".cff", ".bib"],
+    ):
+        return True
+
+    # Look for a README.md file and read its content
+    if (
+        file_content := find_and_read_file(repo=repo, filename="readme.md")
+    ) is not None:
+        # Check for an H2 heading indicating a citation section
+        if any(
+            check_string in file_content
+            for check_string in ["## Citation", "## Citing"]
```

Comment:
perhaps "Cite" is also another option? Also, could it be too restrictive to require this to be in an H2 heading? I suppose you will reduce false positives this way and markdown rendering is a more sustainable pattern... 🤔

Reply:
Thanks and good points! I'll add what you mentioned about `Cite`. I feel that the H2 heading is a good indicator that there's a section dedicated citations - it could be that this is over-permissive. Other things we could search for might be "How to cite" and perhaps a DOI badge of some kind, or perhaps underlined headers from RST. Older repositories might include README's as text files with no additional formatting - these especially would be difficult to validate.